### PR TITLE
Bug/session timeout/post submission fix

### PR DESCRIPTION
**Pull request by neil-stephen-mcgonigle**
- added session alive pre-check for session timeout modal instantiation

### CODE REVIEW

File: src/js/scripts.js [hunk 1]
```diff
@@ -79,7 +79,10 @@ import msToMinutesAndSeconds from '../modules/modal-timeout/utils/msToMinutesAnd
             }
         }
 
-        if (modalElements.length) {
+        sessionData = getSessionData();
+        const sessionEnded = !sessionData.alive;
```
Comment (armoj) [inline src/js/scripts.js:83]: remove this

**Review by armoj — CHANGES_REQUESTED** [review]

File: kube_deploy/Dev/deploy.yml [hunk 2]
```diff
@@ -79,6 +79,11 @@ spec:
                 secretKeyRef:
                   name: cica-web-secrets
                   key: cw_os_places_api_key
+            - name: CW_SESSION_DURATION
```
Comment (armoj) [inline kube_deploy/Dev/deploy.yml:82]: Add this is a separate commit to make it clear the intent behind this change

File: .gitignore [hunk 3]
```diff
@@ -69,3 +69,6 @@ typings/
 .vscode
 
 .config
+
+
+.cache
```
Comment (armoj) [inline .gitignore:74]: Go for it. Maybe have it in a separate commit to readability

File: src/js/scripts.js [hunk 4]
```diff
@@ -80,164 +80,162 @@ import msToMinutesAndSeconds from '../modules/modal-timeout/utils/msToMinutesAnd
...
+            if (!sessionAlive) {
+                return;
+            }
+
             window.document.addEventListener('visibilitychange', () => {
                 documentVisible = window.document.visibilityState === 'visible';
                 if (documentVisible) {
-                    sessionData = getSessionData();
-                    checkShouldOpenModal();
+                    if (sessionAlive) {
```
Comment (armoj) [inline src/js/scripts.js:95]: remove this check

**Review by armoj — CHANGES_REQUESTED** [review]

File: src/js/scripts.js [hunk 5]
```diff
@@ -79,7 +79,10 @@ import msToMinutesAndSeconds from '../modules/modal-timeout/utils/msToMinutesAnd
             }
         }
 
-        if (modalElements.length) {
+        sessionData = getSessionData();
```
[the remark this answers is not in the document]
Comment (neil-stephen-mcgonigle) [reply src/js/scripts.js]: Refactored the code and made new improvements which we discussed yesterday.

File: src/js/scripts.js (hunk 1 above)
Comment (neil-stephen-mcgonigle) [reply src/js/scripts.js]: Refactored the code and made new improvements which we discussed yesterday.

File: src/js/scripts.js [hunk 6]
```diff
@@ -79,7 +79,10 @@ import msToMinutesAndSeconds from '../modules/modal-timeout/utils/msToMinutesAnd
...
+        const sessionEnded = !sessionData.alive;
+
+        if (modalElements.length && !sessionEnded) {
```
[the remark this answers is not in the document]
Comment (neil-stephen-mcgonigle) [reply src/js/scripts.js]: Refactored the code and made new improvements which we discussed yesterday.

File: kube_deploy/Dev/deploy.yml (hunk 2 above)
Comment (neil-stephen-mcgonigle) [reply kube_deploy/Dev/deploy.yml]: this was a goof

File: src/js/scripts.js (hunk 4 above)
Comment (neil-stephen-mcgonigle) [reply src/js/scripts.js]: Refactored the code and made new improvements which we discussed yesterday.

File: .gitignore (hunk 3 above)
Comment (neil-stephen-mcgonigle) [reply .gitignore:74]: done

File: test/livechat-helper.test.js [hunk 7]
```diff
@@ -60,7 +60,9 @@ describe('Live Chat Helper', () => {
             it('should have a list of future bank holidays', () => {
                 // eslint-disable-next-line global-require
                 const bankHolidays = require('../index/bank-holidays.json');
-                const today = new Date();
+                // this functionality is curently disabled
```
Comment (armoj) [inline test/livechat-helper.test.js:63]: I would either remove this test and updated the jest config coverage, or update the bank holidays. voiding this test seems like a bad omen.

https://www.gov.uk/bank-holidays.json

**Review by armoj — CHANGES_REQUESTED** [review]

File: test/livechat-helper.test.js (hunk 7 above)
Comment (neil-stephen-mcgonigle) [reply test/livechat-helper.test.js:63]: Fair play, I actually misread the behaviour of the test.

File: test/livechat-helper.test.js [hunk 8]
```diff
@@ -60,6 +60,13 @@ describe('Live Chat Helper', () => {
             it('should have a list of future bank holidays', () => {
                 // eslint-disable-next-line global-require
                 const bankHolidays = require('../index/bank-holidays.json');
+                // when this test fails update the bank-holidays list for Scotland
+                // and rerun the test
+                // https://www.gov.uk/bank-holidays.json
+                // this test ensures that upcoming bank holidays are present in the bank-holidays.json
+                // and if the live chat functionality is active it will be deactivated for
+                // those Scottish bank holidays present within the bank-holidays.json
+                // this test acts as a reminder to update the bank-holidays.json
```
Comment (armoj) [inline test/livechat-helper.test.js:69]: (☞ﾟヮﾟ)☞

**Review by armoj — APPROVED** [review]

**Pull request merged**